lesson19/src/js/slider.js

```diff
@@ -0,0 +1,210 @@
+    button.setAttribute("aria-label", `${element}`);
+    element === 'previous' && (button.disabled = 'true');
+    fragment.appendChild(button);
+  });
+
+  const slideArea = document.getElementById('js-mainvisual-images');
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagination(item) {
+  const pagination = createAttributedElements("div", {class:"slider-pagination"});
+  const fragment = document.createDocumentFragment();
+  const current = document.createElement("span");
+  current.id = "js-current";
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = createAttributedElements("span", {class:"separation"});
+  separation.textContent = "/";
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.id = "js-total";
+  total.textContent = item;
+  fragment.appendChild(total);
+
+  const textArea = document.getElementById("js-mainvisual-textarea");
+  textArea.appendChild(pagination).appendChild(fragment);
+}
+
+function renderPaginationBullet(item){
+  const paginationArea = createAttributedElements("div",{class:"mainvisual-paginationbullet"});
+  paginationArea.id = 'js-pagination';
+  const fragment = document.createDocumentFragment();
+  for(let i = 0; i < item; i++){
+    const paginationBullet = createAttributedElements("span", {class:"paginationbullet__bar"});
+    paginationBullet.style.width = "calc(width / item)";
+    paginationBullet.setAttribute ("role", "button");
+    paginationBullet.dataset.paginationIndex = i + 1;
+    fragment.appendChild(paginationBullet);
+
+    i === 0 && paginationBullet.setAttribute("aria-current","true");
+
+  }
+  const slideImageArea = document.getElementById("js-mainvisual-images-inner");
+  slideImageArea.appendChild(paginationArea).appendChild(fragment);
+}
+
+function addNoimage(){
+  const slideImageArea = document.getElementById("js-mainvisual-images-inner");
+  const sliderImage = createAttributedElements("img", {class:"slider__image"});
+  sliderImage.src = './images/slide-image01.png'
+  sliderImage.width = 800;
+  sliderImage.height = 800;
+  sliderImage.style.opacity = 1;
+  slideImageArea.appendChild(sliderImage);
+}
+
+function changeActivePagination(currentIndex) {
+  const paginationCurrent = document.getElementById("js-current");
+  paginationCurrent.textContent = currentIndex;
+
+}
+
+function changeActiveImage(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  const changeSlide = document.querySelector(`[data-slide-index="${target}"]`);
+  changeSlide.classList.add("is-active");
+}
+
+function changeActivePaginationBullet(target){
+  const paginationBullet = document.querySelector('[aria-current="true"');
+  paginationBullet.setAttribute('aria-current', false);
+  const pagiNations = document.querySelector(`[data-pagination-index="${target}"]`);
+  pagiNations.setAttribute("aria-current", true);
```

Comment:
Q
Why is only n capitalized? Since "pagination" is one word, it might as well be all lowercase.

Reply:
Thank you!
This has been fixed.

[f9334f0](https://github.com/nyattsu72/hanson-frontend-cource/pull/27/commits/f9334f09182730dd2e7fad406dd5989ad86005a8)